### PR TITLE
Abstract common service functions

### PR DESCRIPTION
**Pull request by KeeyanGhoreshi**
Basically all the post processing steps from both services have been moved to another, single file.  The processing function is now called on the return value of the abstract service function call that all services will have, which effectively makes the new post processing functionality available for any arbitrary future service.  The only caveat is that services need to have the output formatted the same way so that the post processing can take place.  This required a bit of reworking of the original static service to match its output with the CLQ service.  

## Submitter:

**Running the Application**

- [ ] Manually tested in Chrome (primary browser for testing)
- [ ] Manually tested in IE/Safari (verify app loads and basic feature testing)
- [ ] Manually tested in Chrome in iPad mode (at minimum, viewing patient record)

**Task**

- [ ] Branch implements task as expected
- [ ] Key application features are updated as needed for task and tested (see: [Key Application Features](https://github.com/FluxNotes/flux/wiki/Key-Application-Features))
- [ ] Task specific definition of done (as described in the Jira task) is met, if applicable.

**Documentation**

- [ ] This pull request describes why these changes were made
- [ ] Recognizes any potential shortcomings/bugs in the description 
- [ ] Pre release script is updated on the [Pre Release Testing Script wiki page](https://github.com/FluxNotes/flux/wiki/Pre-Release-Testing-Script) (add/remove single features to test, do not run through full script)
- [ ] Update the list of key features with new features on the [Key Application Features wiki page](https://github.com/FluxNotes/flux/wiki/Key-Application-Features)
- [ ] Document any new APIs/extension points
- [ ] Additional technical components and libraries are documented and added to [wiki](https://github.com/FluxNotes/flux/wiki/About-Flux-Notes) as needed

**Code Quality**

- [ ] Code style guide is followed (see: [Code Style Guide](https://github.com/FluxNotes/flux/wiki/JavaScript-and-HTML-Code-Style-Guide))
- [ ] Code is commented

**Tests**

- [ ] Existing tests passed
- [ ] Added backend tests for new functionality added

## Reviewers:

- Check that code is maintainable and reusable. Code reuses existing code and infrastructure where appropriate.
- Check the PR and code accomplishes the task's purpose.
- Check that new tests appropriately test the new code, including edge cases. Other existing tests pass.
- Verify Pull Request checklist is correctly completed by submitter.
- Try to break the code

### CODE REVIEW

File: src/mcode-pilot/utils/serviceResultsProcessing.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+            let survivalRate = parseInt(outcome.survivalRate)/12;
+            row.survivorsPerYear[survivalRate] = outcome.total;
+        });
+        return row;
+    }).filter((x) => x);
+}
+
+/* Process the entire service response. */
+function processResults(data) {
+    console.log(data);
```
Comment (dehall) [inline src/mcode-pilot/utils/serviceResultsProcessing.js]: please remove this console.log

File: src/actions/mcode.js [hunk 2]
```diff
@@ -58,7 +59,8 @@ function processSimilarPatientOutcomes() {
     return (dispatch, getState) => {
         const { similarPatientProps } = getState().mcode;
         const service = new ServiceManager().getService('outcomes');
-        return service.processSimilarPatientOutcomes(similarPatientProps).then(results => {
+        return service.processSimilarPatientOutcomes(similarPatientProps).then(unprocessedResults => {
```
Comment (dehall) [inline src/actions/mcode.js]: It's a little strange to see a function `processSimilarPatientOutcomes` get called, and then what it returns is called `unprocessedResults`. Can the names be clarified here?

File: src/mcode-pilot/utils/serviceResultsProcessing.js [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+expected by the Compass UI components. */
+function generateOutcomeData(data) {
+    return data.map((item) => {
+        if (item.total === 0 || !item.sufficiency) {
+            return null;
+        }
+        let row = initializeTreatmentData(generateTreatmentDisplayName(item.treatments), item.treatments);
+        row.totalPatients = item.total;
+
+        if (item.sideEffects !== undefined) {
```
Comment (dehall) [inline src/mcode-pilot/utils/serviceResultsProcessing.js]: Recommend using `item.sideEffects != undefined` or `!= null` here (and similar comments in other places where `!== undefined` is currently used. This variable may never be `null` in the code as written today but it's safe to be a little defensive here.

File: src/mcode-pilot/services/outcomes/CLQOutcomesService.js [hunk 4]
```diff
@@ -1,35 +1,22 @@
...
-            }
-        });
-        return single ? found[0] : found;
-    }
-
     /* Build the CLQ demograpchics filter section based off of the Compass filter criteria
      */
-    buildDemographicsFilter(similarPatientProps) {
-        let demoProps = similarPatientProps.demographic.options;
+    buildDemographicsFilter(filterOptions) {
```
Comment (dehall) [inline src/mcode-pilot/services/outcomes/CLQOutcomesService.js]: This variable is called `filterOptions` but it's not actually an instance of the `FilterOptions` class, it's the result of `FilterOptions.getAllActiveValuesByMcodeElement()`. Can you make that more clear? Or possibly just add some JSDoc to clarify what this is.  Same comment for `buildDiagnosisFilter` and `buildTumorMarkersFilter`

File: src/mcode-pilot/utils/serviceResultsProcessing.js (hunk 3 above)
Comment (Dtphelan1) [reply src/mcode-pilot/utils/serviceResultsProcessing.js]: Can we use `_.isEmpty` or something along those lines?

**Comment by rdingwell** [comment]
Just for clarity as I go through this.  This branch appears to be started from a branch that is not merged into master yet, is that correct.  IE the filter options class branch.

**Comment by KeeyanGhoreshi** [comment]
Yes

**Pull request merged**